### PR TITLE
MSC3904: Room version 10 as the default room version

### PR DESCRIPTION
**Pull request by FSG-Cat**
This is a derivative of MSC3589 that bumps us up to v10 as default.

[Rendered](https://github.com/FSG-Cat/matrix-spec-proposals/blob/Room-Version-10-Default/proposals/3904-room-version-10-as-a-default.md)

Signed-off-by: Catalan Lover <catalanlover@protonmail.com>

PR to implement in Dendrite: https://github.com/matrix-org/dendrite/pull/2933
PR to implement in Synapse: https://github.com/matrix-org/synapse/pull/14111

[Spec PR](https://github.com/matrix-org/matrix-spec/pull/1397)

[FCP tickyboxes](https://github.com/matrix-org/matrix-spec-proposals/pull/3904#issuecomment-1361620333)

### CODE REVIEW

File: proposals/3904-room-version-10-as-a-default.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Some server implementations, like Synapse, support configurable default room versions: servers which
+have set this flag will not necessarily be affected by this change.
+
+As of writing (2022-10-09) Synapse and Dendrite both have supported Room version 10 since 2022-08-02
+when Synapse in 1.64.0 added its support. Dendrite added its support back in 2022-06-01. This leaves
+Conduit as the only major implementation to lack v10 support. 
+
+Conduit status for v10 is not implemented as of (2022-10-09) but has been reported that its getting close
+by Conduit developer Timo. Ruma has it implemented and the Catalyst Conduit fork has v10 working for about
+as long as Synapse has had its implementation based on the memory of the Author of this MSC.
```
Comment (turt2live) [inline proposals/3904-room-version-10-as-a-default.md:31]: Citation needed on the Timo reference. Unfortunately, Catalyst is not really a suitable implementation for this sort of MSC.

**Review by turt2live — COMMENTED** [review]
I'm a bit concerned that it's still too early to make v10 the default, which is why an MSC from the SCT hasn't shown up fwiw.

File: proposals/3904-room-version-10-as-a-default.md (hunk 1 above)
Comment (FSG-Cat) [reply proposals/3904-room-version-10-as-a-default.md]: I added a link to the message in the spec room where Timo talks about that a new release is on its way. This
was interperted by me that we might be looking at getting a v10 supporting conduit relatively soon. And i pointed out Catalyst to show an example where Rumas v10 implementation has been used in the real world successfuly.

**Comment by turt2live** [comment]
@Mikaela please ask these sorts of questions in the #matrix-spec:matrix.org room rather than on the MSCs.

**Review by turt2live — APPROVED** [review]
conduit is reported to have v10 support as of [v0.5.0](https://gitlab.com/famedly/conduit/-/tags/v0.5.0), however I have not validated that.

Enough time has passed where this MSC should not be blocked, though.

**Comment by turt2live** [comment]
@mscbot fcp merge

**Comment by turt2live** [comment]
Spec PR: https://github.com/matrix-org/matrix-spec/pull/1397

**Comment by turt2live** [comment]
Merged 🎉

**Pull request merged**